Point2D point
uint32 color